mono/metadata/w32handle.c

```diff
@@ -143,6 +144,8 @@ mono_w32handle_trylock (MonoW32Handle *handle_data)
 void
 mono_w32handle_unlock (MonoW32Handle *handle_data)
 {
+	if (!handle_data)
```

Comment:
This cannot transparently fail. Check in the caller that we do not unlock a invalid handle.